src/aggregate/aggregate_exec.c

```diff
@@ -912,6 +912,9 @@ static int execCommandCommon(RedisModuleCtx *ctx, RedisModuleString **argv, int
   if (!IsInternal(r) || IsProfile(r)) {
     // We currently don't need to measure the time for internal and non-profile commands
     r->initClock = clock();
+  }
+
+  if (r->qiter.isProfile) {
```

Comment:
Have you considered consolidating all our clocks to use the `struct timespec` instead of the current redundancy? Obviously it is caused due to our mixed usage of the Wall-clock and the CPU time (via the `clock()` function)?
This will actually be great for us, since it will guarantee a minimal precision of nanoseconds, remove flakiness that we may receive from the current CPU time clock, and unify our clock mechanisms to the more precise option.

Reply:
We don't have much of the CPU counter-clocks.